File: spring_java_orm_sample/src/main/java/com/example/demo/api/request/PurchaseInvoiceDetailCreateRequest.java
package com.example.demo.api.request;

import com.example.demo.api.validator.Isbn;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Value;

@Value
public class PurchaseInvoiceDetailCreateRequest {
    @Schema(description = "仕入伝票明細ISBN", type = "string", example = "0000000000001")
    @NotNull
    @Isbn
    String purchaseInvoiceDetailIsbn;

    @Schema(description = "仕入伝票明細単価")
    @Max(10000)
    @Min(1)
    @NotNull
    Integer purchaseInvoiceDetailUnitPrice;

    @Schema(description = "仕入伝票明細数量")
    @Max(1000)
    @Min(1)
    @NotNull
    Integer purchaseInvoiceDetailQuantity;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/api/request/PurchaseInvoiceDetailCreateRequest.kt
package com.example.demo.api.request

import com.example.demo.api.annotation.Isbn
import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull

data class PurchaseInvoiceDetailCreateRequest(
    @field:Schema(description = "仕入伝票明細ISBN", type = "string", example = "0000000000001")
    @field:NotNull
    @field:Isbn
    val purchaseInvoiceDetailIsbn: String?,

    @field:Schema(description = "仕入伝票明細単価")
    @field:Max(10000)
    @field:Min(1)
    @field:NotNull
    val purchaseInvoiceDetailUnitPrice: Int?,

    @field:Schema(description = "仕入伝票明細数量")
    @field:Max(1000)
    @field:Min(1)
    @field:NotNull
    val purchaseInvoiceDetailQuantity: Int?
)
